Segmentation violation thrown when using Bearer authentication

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately see https://github.com/kubernetes/kube-state-metrics/blob/master/SECURITY.md
-->

**What happened?**:

When the adapter starts, it crashes with the following error when attempting to connect to Prometheus (actually Thanos Querier) when I enable the ``--prometheus-token-file`` flag.
(the following line is the cause: https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/cmd/adapter/adapter.go#L116)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1478685]

goroutine 558 [running]:
k8s.io/client-go/transport.(*bearerAuthRoundTripper).RoundTrip(0xc00019e510, 0xc000369e00, 0xc00019e510, 0x0, 0x0)
	/go/pkg/mod/k8s.io/client-go@v0.22.0/transport/round_trippers.go:298 +0x265
net/http.send(0xc000369e00, 0x237b3a0, 0xc00019e510, 0x0, 0x0, 0x0, 0xc000a2a0a0, 0x203000, 0x1, 0x0)
	/usr/local/go/src/net/http/client.go:251 +0x454
net/http.(*Client).send(0x3195c40, 0xc000369e00, 0x0, 0x0, 0x0, 0xc000a2a0a0, 0x0, 0x1, 0xc000369e00)
	/usr/local/go/src/net/http/client.go:175 +0xff
net/http.(*Client).do(0x3195c40, 0xc000369e00, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:717 +0x45f
net/http.(*Client).Do(...)
	/usr/local/go/src/net/http/client.go:585
sigs.k8s.io/prometheus-adapter/pkg/client.(*httpAPIClient).Do(0xc00000cdc8, 0x23bbf08, 0xc000052098, 0x20be845, 0x3, 0x20cb2b3, 0xe, 0xc000d23260, 0x0, 0x0, ...)
	/go/src/sigs.k8s.io/prometheus-adapter/pkg/client/api.go:64 +0x2eb
sigs.k8s.io/prometheus-adapter/pkg/client/metrics.(*instrumentedGenericClient).Do(0xc0004455c0, 0x23bbf08, 0xc000052098, 0x20be845, 0x3, 0x20cb2b3, 0xe, 0xc000d23260, 0x0, 0x0, ...)
	/go/src/sigs.k8s.io/prometheus-adapter/pkg/client/metrics/metrics.go:69 +0x1d6
sigs.k8s.io/prometheus-adapter/pkg/client.(*queryClient).Series(0xc00027a2e0, 0x23bbf08, 0xc000052098, 0x17e357cedf1, 0x0, 0xc00072d280, 0x1, 0x1, 0xc00002b8c0, 0xc000556f30, ...)
	/go/src/sigs.k8s.io/prometheus-adapter/pkg/client/api.go:163 +0x1f7
sigs.k8s.io/prometheus-adapter/pkg/custom-provider.(*cachingMetricsLister).updateMetrics.func1(0xc00044fb30, 0x17e357cedf1, 0xc0007224b0, 0x45, 0xc000778ea0, 0xc000778e40)
	/go/src/sigs.k8s.io/prometheus-adapter/pkg/custom-provider/provider.go:259 +0xe2
created by sigs.k8s.io/prometheus-adapter/pkg/custom-provider.(*cachingMetricsLister).updateMetrics
	/go/src/sigs.k8s.io/prometheus-adapter/pkg/custom-provider/provider.go:258 +0x9da
```


**What did you expect to happen?**:

I expected to see Prometheus Adapter successfully connect to Prometheus/Thanos.

**Please provide the prometheus-adapter config**:
<details open>
<summary>prometheus-adapter config</summary>

![image](https://user-images.githubusercontent.com/2869798/148580217-e0ef051a-8f56-401a-ad03-f9fe1da2369e.png)
I also tried to directly mount the token using ``--prometheus-token-file=/var/run/secrets/kubernetes.io/serviceaccount/token`` but same result.

The config is pretty standard, I use Helm for deployment, here is my values.yaml file:

![image](https://user-images.githubusercontent.com/2869798/148580599-0d707f6b-a1a6-4a2d-986d-67ac221e7fb8.png)

</details>

**Anything else we need to know?**:

If this flag cannot be used how can I set up Authentication using Bearer token? I have a working Grafana connecting on the same Thanos as a datasource using the Bearer Token method, but I keep getting 403 Forbidden errors on the Adapter.
I also tried ``--prometheus-header`` parameter but it does nothing (still 403).

**Environment**:
- prometheus-adapter version: v0.9.1
- prometheus version: Thanos Querier 0.20.2
- Kubernetes version (use `kubectl version`): OpenShift 4.8.7 (k8s v1.21)
- Cloud provider or hardware configuration: On-Prem amd64
- Other info: -

--- a/cmd/adapter/adapter.go
+++ b/cmd/adapter/adapter.go
@@ -119,7 +119,11 @@ func (cmd *PrometheusAdapter) makePromClient() (prom.Client, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to read prometheus-token-file: %v", err)
 		}
-		httpClient.Transport = transport.NewBearerAuthRoundTripper(string(data), httpClient.Transport)
+		wrappedTransport := http.DefaultTransport
+		if httpClient.Transport != nil {
+			wrappedTransport = httpClient.Transport
+		}
+		httpClient.Transport = transport.NewBearerAuthRoundTripper(string(data), wrappedTransport)
 	}
 	genericPromClient := prom.NewGenericAPIClient(httpClient, baseURL, parseHeaderArgs(cmd.PrometheusHeaders))
 	instrumentedGenericPromClient := mprom.InstrumentGenericAPIClient(genericPromClient, baseURL.String())

